src/api/v2/v2AzureResourcesApiImplementation.ts

```diff
@@ -1,14 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
 import * as vscode from 'vscode';
-import { BranchDataProviderManager } from '../../tree/v2/providers/BranchDataProviderManager';
-import { ApplicationResourceProviderManager } from './providers/ApplicationResourceProviderManager';
-import { ApplicationResource, ApplicationResourceProvider, BranchDataProvider, ResourcePickOptions, V2AzureResourcesApi, WorkspaceResource, WorkspaceResourceProvider } from './v2AzureResourcesApi';
+import { ApplicationResourceBranchDataProviderManager } from '../../tree/v2/application/ApplicationResourceBranchDataProviderManager';
+import { WorkspaceResourceBranchDataProviderManager } from '../../tree/v2/workspace/WorkspaceResourceBranchDataProviderManager';
+import { ApplicationResourceProviderManager } from './ApplicationResourceProviderManager';
+import { ApplicationResource, ApplicationResourceProvider, BranchDataProvider, ResourceModelBase, ResourcePickOptions, V2AzureResourcesApi, WorkspaceResource, WorkspaceResourceProvider } from './v2AzureResourcesApi';
+import { WorkspaceResourceProviderManager } from './WorkspaceResourceProviderManager';
 
 export class V2AzureResourcesApiImplementation implements V2AzureResourcesApi {
     public static apiVersion: string = '2.0.0';
 
     constructor(
-        private readonly branchDataProviderManager: BranchDataProviderManager,
-        private readonly resourceProviderManager: ApplicationResourceProviderManager) {
+        private readonly branchDataProviderManager: ApplicationResourceBranchDataProviderManager,
+        private readonly resourceProviderManager: ApplicationResourceProviderManager,
```

Comment:
Nit: Since the resources are provided before the BDP kicks in, I'd flip the arguments around. i.e.
```
resourceProviderManager,
branchDataProviderManager,
workspaceResourceProviderManager,
workspaceResourceBranchDataProviderManager
```

Reply:
I suppose; I was just going in alphabetical order.  Updated.